### ISSUE TITLE
template PYPI_SOURCE cannot be used to download python extensions

### ISSUE THREAD

**Issue by pescobar** [easybuilders/easybuild-framework#1597]
I am writing a python bundle for scikit-bio and I tried to add this extension

```
exts_list = [
_genutils', '0.1.0', {
        'source_urls': [PYPI_SOURCE],
    }),
]
```

but the download fails because the url uses the main download url in the easyconfig

```
runpy.filetools WARNING Download of https://pypi.python.org/packages/source/s/scikit-bio/ipython_genutils-0.1.0.tar.gz to /export/soft/source/s/scikit-bio/extensions/ipython_genutils-0.1.0.tar.gz failed,
```

it would be nice if `PYPI_SOURCE` and `PYPI_LOWER_SOURCE` could be used for python extensions. It would simplify a lot writing python bundles

**Comment by serverhorror** [comment]
Could it be that this is the culprit:

* https://github.com/hpcugent/easybuild-framework/blob/027438dffb67121e9feb2a4764a72ccaa338f1c7/easybuild/framework/easyconfig/templates.py#L102-L105

Current:

```python
    ('PYPI_SOURCE', 'https://pypi.python.org/packages/source/%(nameletter)s/%(name)s',
     'pypi source url'),  # e.g., Cython, Sphinx
    ('PYPI_LOWER_SOURCE', 'https://pypi.python.org/packages/source/%(nameletterlower)s/%(namelower)s',
     'pypi source url (lowercase name)'),  # e.g., Greenlet, PyZMQ
```
Should be:
```python
    ('PYPI_SOURCE', 'https://files.pythonhosted.org/packages/source/%(nameletter)s/%(name)s',
     'pypi source url'),  # e.g., Cython, Sphinx
    ('PYPI_LOWER_SOURCE', 'files.pythonhosted.org/packages/source/%(nameletterlower)s/%(namelower)s',
     'pypi source url (lowercase name)'),  # e.g., Greenlet, PyZMQ
```

I gave it a quick try but it seems that `nameletter*` is never actually resolved and I couldn't find why this would be happening as it will be populated for the actual easyconfig but not for the entries in `exts_list`.

**Comment by boegel** [comment]
@serverhorror The actual URLs are not the problem. For PyPI specifically, we translate the URLs obtained through `PYPI_SOURCE` via `derive_alt_pypi_url` to an actual download URL (see https://bitbucket.org/pypa/pypi/issues/438 why).

The problem is indeed that templates like `%(name)s` are not being resolved correctly for extensions, since `name` is still the name of the 'parent' software. I'm not sure if there's a solution for that though, since it's hard to tell whether you want the value to resolve as the extension name of the parent software's name, it depends on the context...

**Comment by serverhorror** [comment]
If that is any help:

I'd expect the templates to refer to the extension not the parent. It just feels natural to me. Otherwise things like `%(nameletter)s` don't make any sense. Think about `Python` with `SQLAlchmey`in `exts_lists`. Now when `nameletter` is `P` how is that related to `SQLAlchemy`, or any other current extension for that matter?

Another reason why I would think the inner scope should be what expands the value: If I do a `PyhtonPackage` in its own file, then the templates will refer to the values from this very EasyConfig not from the Parent. I'd expect the same behaviour from `exts_list` because in my mind it just runs thru a list of `*Package` with a shorthand syntax.

(Merely input from what I think a user would expect. User would be me in that case).

**Comment by boegel** [comment]
fixed in #2852
